Add detail page for artifacts and job.

splattael opened the pull request:
## Warning: Untested code ahead

I've just added detail pages for builds and jobs.
### Repo

Each build displayed on the repo page is now linked to the specific build (e.g. #13 is linked).

![Repo](http://s3.amazonaws.com/imgly_production/6321303/original.png)
### Build details

The new build page display all jobs for this build. Each job links to the specific job detail page.

![Builds](http://s3.amazonaws.com/imgly_production/6321324/original.png)
### Job details

Display job details including job build log.

![Job](http://s3.amazonaws.com/imgly_production/6321333/original.png)
### Finally

The code in this PR is not tested and needs your feedback.

If you like the idea of this PR I'm happy to add specs and such.

Commit:
Add detail page for artifacts and job.

splattael commented:
This PR is related #3

sarahhodne commented:
Awesome, I'll merge this in locally and add some tests to it before I push it.

sarahhodne commented:
Just read the last line of your pull request. If you want to add specs, please do. I'll look over the code once more and give you some feedback.

sarahhodne commented on lib/travis/lite/templates/job.mustache:
```diff
@@ -0,0 +1,23 @@
+    <li>Status: <span class="label label-{{status_class}}">{{status}}</li>
+    <li>Branch: {{branch}}</li>
+    <li>Language: {{config_label}}</li>
+    <li>Compare <a href="{{compare_url}}">on github</a></li>
```
This should be "GitHub" and not "github". See @kneath's [tweet](https://twitter.com/kneath/status/275841609151700992).

kneath replied:
:heart_eyes_cat:

sarahhodne commented:
Code looks good. There are some naming issues, but they're mostly in the code I created earlier, so I'll just update everything later to remain consistent.

Commit:
Fix GitHub's name.

Commit:
Add spec for BuildFetcher#fetch_build_for_slug

Commit:
Rename build_fetche to build_fetcher.

Commit:
It's actually one build object.

[ci skip]

Commit:
Add specs for JobFetcher.

Commit:
DRY up build status. Still needs some love.

splattael commented:
@henrikhodne I've added some specs.

I've moved `format_build_status` and friends to `Layout`. Repos are now being marking yellow when running.

You talked about "naming issues". Which one? :) I guess I misnamed `Matrix` which is might be called `Artifact` instead.

sarahhodne commented:
@splattael The naming issues I was actually the `API` methods not having the "type" of resource being returned in the name, but I must've misread the code since I don't see it anymore.

`Matrix` should be `Matrix` in my opinion, `Artifact` is a completely different thing (currently the only artifacts are logs).

Thanks for the PR!

sarahhodne commented:
Just deployed this. Thanks again!
